Two MooseDocs quality-of-life improvements

cticenhour opened the pull request:
Closes #31263
Closes #31264

@NamjaeChoi, @bwspenc, @hugary1995 - for the portion of this relevant to our discussion on Slack, sample output for an improved missing markdown file error message reads:

```
/MySystem/MyObject is missing a markdown file.
  The page should be located at framework/doc/content/source/mysystem/MyObject.md.
  A stub page can be created using the './moosedocs.py generate' command.
  This error can also be encountered if a new markdown page is not known by git ls-files.
  In this case, please doublecheck that the file has been committed or that a 'git add' has
    been performed for any new documentation files!"
Control this message using: 'log_missing_markdown'
```

Let me know if you have any comments on this. 

While I am here, I might as well check off another quality-of-life item that has been bugging me for ages - adding some optional progress output to the `git_init_submodule` method.

cticenhour committed:
Add progress indicator to git_init_submodule method in mooseutils

Enable progress for MooseDocs large_media pull

Refs #31264

cticenhour committed:
Improve missing markdown error message in MooseDocs 'check'

Refs #31263

hugary1995 commented on python/moosesqa/check_syntax.py:
```diff
@@ -66,8 +66,11 @@ def _check_node(node, file_cache, object_prefix, syntax_prefix, logger):
         msg = "{} is missing a markdown file.\n".format(node.fullpath())
-        msg += "  The page should be located at {}\n".format(md_path)
-        msg += "  A stub page can be created using using the './moosedocs.py generate' command"
+        msg += "  The page should be located at {}.\n".format(md_path)
+        msg += "  A stub page can be created using the './moosedocs.py generate' command.\n"
+        msg += "  This error can also be encountered if a new markdown page is not known by git ls-files.\n"
+        msg += "  In this case, please doublecheck that the file has been committed or that a 'git add' has \n"
+        msg += "    been performed for any new documentation files!"
```
I'd say something like
```
msg += "In this case, please track the new markdown page using `git add {}`.".format(node.fullpath())
```

I'd also favor a smarter error message
```
if Path(node.fullpath()).exists():
  msg += "  We noticed that the page exists locally but has not been tracked by git."
  msg += "  If you intend to use this page, please track it using `git add {}`.".format(node.fullpath())
```

cticenhour replied:
For the case when the file is not detected, the previous message still applies. Let me know if you want to make further tweaks.

hugary1995 replied:
Perfect!

cticenhour committed:
Make missing markdown error message smarter

Check to see if the file exists locally, and determine appropriate message to use based on that status.

Refs #31263

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3118681/), step Docs: sync website on 4987275 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31265/site)

This comment will be updated on new commits.
